accounts-db/src/accounts_db.rs

```diff
@@ -334,13 +337,17 @@ impl CurrentAncientAppendVec {
         }
     }
 
+    /// Create ancient append vec for a slot
+    ///     min_bytes: the new append vec needs to have at least this capacity
     #[must_use]
     fn create_ancient_append_vec<'a>(
         &mut self,
         slot: Slot,
         db: &'a AccountsDb,
+        min_bytes: usize,
     ) -> ShrinkInProgress<'a> {
-        let shrink_in_progress = db.get_store_for_shrink(slot, get_ancient_append_vec_capacity());
+        let size = get_ancient_append_vec_capacity().max(min_bytes as u64);
```

Comment:
@HaoranYi if I get rid of .max* here on this line, the only test that fails is one that uses this parameter. Apparently no other tests are testing when we try to write too much in overflow. Can you please add a test that would fail without this .max?

Reply:
Yes. Added a new test to cover this.